### PR TITLE
Feature/cerberus metrics

### PR DESCRIPTION
**Pull request by fieldju**
- Added module / CloudFormation template to create a SNS Topic for publishing Cerberus Metrics and KPIs.
- Configured backup lambda to publish KPIs to metrics topic.
- Created Reference Lambda for process metric messages.

### CODE REVIEW

**Comment by tlisonbee** [comment]
Seems like there should be a corresponding update to the website (creating-an-environment.md)

File: README.md [hunk 1]
```diff
@@ -12,6 +12,17 @@ This project contains the serverless components that can be used with Cerberus.
 * [Cerberus Cross Region Backup Lambda](cerberus-cross-region-backup-lambda/README.md) - Serverless function for for making complete backups of the data in Cerberus (CMS / Vault) and backing it up in a different region encrypted with KMS. With this function enabled, Cerberus operators can use the CLI to complete data restores in new regions or the original region.
 * [Cerberus Lambda VPC](cerberus-lambda-vpc/README.md) - Cloudformation to create a VPC with EIBs and NATs so that Cerberus operators can run lambdas with predictable IP addresses. The backup lambda should run in this VPC so that the IPs can be white listed to avoid rate limiting.
 
+## Profiles
+
+**Read this first**
+
+This project is configured to load properties from environment specific profiles. in [profile/](profile) 
+you will see an [example.properties](profile/example.properties). Before deploying any of this serverless components
```
Comment (tlisonbee) [inline README.md]: The writing is a little messy here: 'in profile', 'this serverless', etc.  Also, avoid informal 'you' and 'we'.

Maybe rewrite as instructions.

File: cerberus-artemis-kpi-lambda/README.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Cerberus Artemis KPI Lambda
```
Comment (tlisonbee) [inline cerberus-artemis-kpi-lambda/README.md]: I don't know that we should call it Artemis, especially if we aren't defining Artemis.  How about 'Cerberus Example KPI Lambda'?

File: cerberus-artemis-kpi-lambda/src/main/groovy/com/nike/cerberus/Handler.groovy [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import com.amazonaws.services.s3.AmazonS3Client
+import com.fieldju.commons.EnvUtils
+import com.fieldju.commons.StringUtils
+import groovy.json.JsonBuilder
+import groovy.json.JsonSlurper
+import org.apache.log4j.Logger
+
+import java.nio.ByteBuffer
+
+class Handler {
```
Comment (tlisonbee) [inline cerberus-artemis-kpi-lambda/src/main/groovy/com/nike/cerberus/Handler.groovy]: How about at least a line or two of class level Javadoc?

File: cerberus-lambda-vpc/README.md [hunk 4]
```diff
@@ -5,5 +5,5 @@ Each AZ has a private and public subnet
 The private subnets connect to the public subnet with NATs and EIBs
 This VPC allows you to deploy Lambdas to this VPC and have predictable IP addresses
 
-For you convenience if you are ok with lambdas running in us-east-1 you can simply run 
-`./gradlew cerberus-lambda-vpc:deploySam --stacktrace` after configuring a global profile./gradlew cerberus-lambda-vpc:deploySam
\ No newline at end of file
+For your convenience if you are ok with lambdas running in us-east-1 you can simply run
```
Comment (tlisonbee) [inline cerberus-lambda-vpc/README.md]: Awkward wording, missing commas.

How about rewriting as instructions?

1. Configure global profile
2. Optionally configure ...
3. Run command ...

File: cerberus-metrics-topic/README.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
+# Cerberus Metrics Topic
+
+This template creates an SNS topic that is used by various components to publish Cerberus metrics to, if enabled.
```
Comment (tlisonbee) [inline cerberus-metrics-topic/README.md]: 'if enabled' is awkward.  How about 'This optional template ... that can be used by various components ...' or similar?

File: README.md (hunk 1 above)
Comment (fieldju) [inline README.md]: - [ ]

File: cerberus-artemis-kpi-lambda/src/main/groovy/com/nike/cerberus/Handler.groovy (hunk 3 above)
Comment (fieldju) [inline cerberus-artemis-kpi-lambda/src/main/groovy/com/nike/cerberus/Handler.groovy]: - [ ] Add class header.

**Pull request merged**